### PR TITLE
feat(Climbing): Move camera with moving climb object

### PR DESCRIPTION
**Pull request by TravisCornelius**

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4d0e54a48566ada65b5487e3bc18c3e5db921285
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/thestonefox/SteamVR_Unity_Toolkit

---
This message was auto-generated by https://gitcop.com

**Comment by mattboy64** [comment]
Looks good to me.

**Comment by thestonefox** [comment]
needs a `git commit --amend` to change the commit message and also to add a commit message body.

Contributing guidelines are here: https://github.com/thestonefox/SteamVR_Unity_Toolkit/blob/master/CONTRIBUTING.md

**Comment by mattboy64** [comment]
Have you tried it with camera safe teleport? I feel like there can be a nasty bug lurking there now that the climbing object can move. Perhaps that position should take this offset into consideration too.

File: Assets/VRTK/Scripts/VRTK_PlayerClimb.cs [hunk 1]
```diff
@@ -240,11 +242,11 @@ private bool IsClimbableObject(GameObject obj)
             return interactObject != null && interactObject.AttachIsClimbObject();
         }
 
-        private void Update()
+        private void LateUpdate()
         {
             if (isClimbing)
             {
-                transform.position = startPosition - (GetPosition(controllerTransform) - startControllerPosition);
+                transform.position = startPosition - (GetPosition(controllerTransform) - startControllerPosition) + (climbingObject.transform.position - startClimbingObjectPosition);
```
Comment (mattboy64) [inline Assets/VRTK/Scripts/VRTK_PlayerClimb.cs]: This is a great first pass, but you're going to run into issues when the climbing object starts rotating. You've taken translation into account here, but rotation probably needs to be added as well.

**Comment by xEvoGx** [comment]
What does this even do? 

( I read '_climbing_' and got intrigued *cough*I'mABrokenRecord*cough*) lol

**Comment by TravisCornelius** [comment]
I was creating a game where you can grab onto a bird's talons and he flies you around.  It was just a mod for the climb object so it would track moving climbable objects.  I didn't take rotation into account for what I needed it for, I'll update when I get some time

**Comment by mattboy64** [comment]
Yeah, it would work great for that. Do you think you'll get to rotation for this PR?

**Comment by lynamd** [comment]
I love this.  I'm working on a rope swinging mechanic, and I'd love to work with VRTK on it rather than roll my own solution.  Probably need to add something to the rigidbody of the PlayerPresence component to simulate momentum as well.  Otherwise you simply fall straight down on ungrab.

Regarding rotation, you should just be able to grab the climbable's start rotation in OnGrabObject and in rotate it in update like so

    transform.rotation = startRotation * (climbingObject.transform.rotation * Quaternion.Inverse(startClimbingObjectRotation));


However, the collider of your player presence will remain rotated when you ungrab, so you might have to add some correcting code in OnUngrabObject.

**Comment by lucboudreau** [comment]
Again, just for reference, I know a lot of people have attempted this, but I've figured out and shared the code here:

https://github.com/thestonefox/VRTK/issues/716

**Comment by mattboy64** [comment]
I created a version with movement but also added rotation for the 3.0 version of VRTK here: #903

**Comment by thestonefox** [comment]
@mattboy64 does your work that was merged supersede this PR?

**Comment by mattboy64** [comment]
I believe so, yes

**Pull request closed without merging**